Support load shader code from node in the plugin ui

yiqunw700 opened the pull request:
Fixes: #1877 
Changed plugin ui to use type "color3" "color4" so when it loads the shader code from node, it knows it is a color.

Added a load shader button to the node id line:
<img width="582" alt="Screenshot 2025-01-17 at 13 35 35" src="https://github.com/user-attachments/assets/98101aec-b112-449c-8757-d33b5afb7e46" />

rylin8 commented on support-figma/extended-layout-plugin/src/shader.ts:
```diff
@@ -23,9 +23,15 @@ const SHADER_UNIFORMS_PLUGIN_DATA_KEY = "shaderUniforms";
+  uniformName: string;
+  uniformType: string;
+  uniformValue: number | Float32Array | RGBA;
+  extras: ShaderExtras | null;
```
Since there are no Rust changes, I guess the application doesn't have access to min/max values. Would that  be useful to have in the future?

rylin8 reviewed (commented):
Why did the unit test, design switcher, variables and tutorial doc dcf files change?

yiqunw700 commented on support-figma/extended-layout-plugin/src/shader.html:
```diff
@@ -602,10 +660,36 @@
+                                if (shaderUniform.extras) {
+                                    createShaderFloatUniform(shaderUniform.uniformName, shaderUniform.uniformValue, shaderUniform.extras.min, shaderUniform.extras.max);
+                                } else {
+                                    createShaderFloatUniform(shaderUniform.uniformName, shaderUniform.uniformValue, 0, Math.max(1, shaderUniform.uniformValue));
```
Done.

yiqunw700 commented on support-figma/extended-layout-plugin/src/shader.html:
```diff
@@ -382,13 +397,21 @@
+    function rgbFloatToHex(rgbColor) {
+        function componentToHex(c) {
+            const hex = Math.round(c * 255).toString(16);
+            return hex.length === 1 ? "0" + hex : hex;
```
Done.

yiqunw700 commented:
> Why did the unit test, design switcher, variables and tutorial doc dcf files change?

Rebased and variables changes are gone. The layout unit test, design switcher and tutorial doc is updated whenever I ran `./dev-scripts/update_dcfs.sh` which uses the `cargo` command. I don't know why. I reverted them manually.
